src/Extension/HeadingPermalink/HeadingPermalinkProcessor.php

```diff
@@ -43,36 +44,36 @@ final class HeadingPermalinkProcessor implements ConfigurationAwareInterface
      */
     private $config;
 
-    public function setConfiguration(ConfigurationInterface $configuration): void
+    public function setEnvironment(EnvironmentInterface $environment): void
     {
-        $this->config = $configuration;
+        $this->config         = $environment->getConfiguration();
+        $this->slugNormalizer = $environment->getSlugNormalizer();
```

Comment:
This also instantiates the slug normalizer, regardless if it's needed.

Reply:
Addressed in my comment below